sinks/gcm/driver.go

```diff
@@ -141,43 +151,52 @@ type metricWriteRequest struct {
 const maxTimeseriesPerRequest = 200
 
 // Pushes the specified metric values in input. The metrics must already exist.
-func (self *gcmSink) StoreTimeseries(input []sink_api.Timeseries) error {
+func (self *gcmSink) StoreTimeseries(inputTimeseriesLists []*[]sink_api.Timeseries) error {
 	// Ensure the metrics exist.
-	for _, entry := range input {
-		metric := entry.Point
-		// TODO: Remove this check if possible.
-		if _, ok := self.exportedMetrics[metric.Name]; !ok {
-			return fmt.Errorf("unable to push unknown metric %q", metric.Name)
+	for _, inputTimeseriesList := range inputTimeseriesLists {
+		if inputTimeseriesList != nil {
+			for _, inputTimeseries := range *inputTimeseriesList {
+				// TODO: Remove this check if possible.
+				if _, ok := self.exportedMetrics[inputTimeseries.SeriesName]; !ok {
+					return fmt.Errorf("unable to push unknown metric %q", inputTimeseries.SeriesName)
+				}
+			}
 		}
 	}
 
 	// Build a map of metrics by name.
 	metrics := make(map[string][]timeseries)
-	for _, entry := range input {
-		metric := entry.Point
-
-		// Use full label names.
-		labels := make(map[string]string, len(metric.Labels))
-		for key, value := range metric.Labels {
-			labels[fullLabelName(key)] = value
-		}
-
-		// TODO(vmarmol): Validation and cleanup of data.
-		// TODO(vmarmol): Handle non-int64 data types. There is an issue with using omitempty since 0 is a valid value for us.
-		if _, ok := metric.Value.(int64); !ok {
-			return fmt.Errorf("non-int64 data not implemented. Seen for metric %q", metric.Name)
+	for _, inputTimeseriesList := range inputTimeseriesLists {
+		if inputTimeseriesList != nil {
```

Comment:
I found the same pattern in a few other files. Can you make the same change in other files as well?